source/javascripts/services/_appService.js.erb

```diff
@@ -217,28 +225,51 @@ angular.module("BitriseWorkflowEditor").service("appService", function($q, reque
 			normalizeSecretConfigs(secretConfigs);
 
 			appService.secrets = _.map(secretConfigs, function(aSecretConfig) {
-				return new Variable(aSecretConfig, Variable.defaultVariableConfig());
+				return new Variable(aSecretConfig, Variable.defaultVariableConfig(), false, false);
 			});
 
 			appService.savedSecrets = angular.copy(appService.secrets);
 		});
 	};
 
+	appService.getSecretEnvValue = function(envVar, requestConfig) {
+		var secretEnvVar = envVar;
+		if (!secretEnvVar && appService.secrets) {
+			return $q.when();
+		}
+
```

Comment:
Check if secret already has value, and immediately resolve with a `$q.when()` if it does.